src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs

```diff
@@ -263,6 +269,51 @@ public BulkInsertOperation(string database, IDocumentStore store, BulkInsertOpti
 
             _generateEntityIdOnTheClient = new GenerateEntityIdOnTheClient(_requestExecutor.Conventions,
                 entity => AsyncHelpers.RunSync(() => _requestExecutor.Conventions.GenerateDocumentIdAsync(database, entity)));
+
+            _streamLock = new AsyncReaderWriterLock();
+            _lastWriteToStream = DateTime.Now;
+            _timer = new Timer( async o => await SendHeartBeat(o).ConfigureAwait(false),
+                null,
+                (int)(StreamWithTimeout.DefaultWriteTimeout.TotalMilliseconds / 3),
+                Timeout.InfiniteTimeSpan.Milliseconds);
```

Comment:
this is overkill, if you would define static
e.g.
```
private static readonly TimeSpan HeartbeatCheckInterval = TimeSpan.FromSeconds(StreamWithTimeout.DefaultWriteTimeout.TotalSeconds / 3);
```

and use it as follows:
```
_timer = new Timer(HandleHeartbeat, null, HeartbeatCheckInterval, HeartbeatCheckInterval);
```

then you do not need to dispose timers, change the intervals, etc
It will just do the check every X seconds

Also this is async void IIRC
```
 async o => await SendHeartBeat(o).ConfigureAwait(false)
```

Reply:
done